GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -884,26 +884,27 @@ IReadOnlyList<GitItemStatus> GetAllChangedFilesWithSubmodulesStatus()
 
             if (doAsync)
             {
-                ThreadHelper.JoinableTaskFactory.RunAsync(() =>
+                return ThreadHelper.JoinableTaskFactory.RunAsync(() =>
                 {
                     return _unstagedLoader.LoadAsync(GetAllChangedFilesWithSubmodulesStatus, onComputed);
-                });
+                }).Task;
```

Comment:
❔ just
```
await _unstagedLoader.LoadAsync(GetAllChangedFilesWithSubmodulesStatus, onComputed);
```

Reply:
Reverted this as it should only be awaited in tests and there is `ThreadHelper.JoinPendingOperations();` for that.